src/effects/use-async-request.js

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+import requestStates from '../state/state-utils/request-states';
+
+const useAsyncRequest = (request, performRequest, deps) => {
```

Comment:
`deps` must be expanded into the actual variables that might change. This is because hooks do static analysis of what parameters get passed, and using varaible expansion `[...deps]` will prevent this. You probably want `url, resource, body` as three specific arguments.